dts: bindings: clock: rpi_pico: add XOSC definition

xudongzheng opened the pull request:
On some boards, XOSC can take longer to stabilize.

xudongzheng commented:
Are there any objections to having the default set to 64? I don't think there is a meaningful effect on the startup process. With 1 as the default, some boards will mysteriously hard fault on reset.

Example boards with `RPI_PICO_XOSC_STARTUP_DELAY_MULTIPLIER` defined:

- https://github.com/search?q=repo%3Araspberrypi%2Fpico-sdk%20PICO_XOSC_STARTUP_DELAY_MULTIPLIER&type=code
- https://github.com/search?q=repo%3Aadafruit%2Fcircuitpython%20PICO_XOSC_STARTUP_DELAY_MULTIPLIER&type=code

soburi commented on drivers/clock_control/Kconfig.rpi_pico:
```diff
@@ -16,4 +16,10 @@ config RPI_PICO_ROSC_USE_MEASURED_FREQ
 	 Instead of the dts value, use the value measured by
 	 the frequency counter as the rosc frequency.
 
+config RPI_PICO_XOSC_STARTUP_DELAY_MULTIPLIER
```
This is a hardware-related parameter.
Could you consider moving to the devicetree definition?
Also, you can add a .yaml file to supporting the xosc.

xudongzheng replied:
Please check the latest changes. It's now configured via a `raspberrypi,pico-xosc` binding.

soburi replied:
Gread Job! Thanks!.

xudongzheng committed:
dts: bindings: clock: rpi_pico: add XOSC definition

This defines raspberrypi,pico-xosc along with a configurable startup
delay multiplier. On some boards, the XOSC takes longer to stabilize.

Signed-off-by: Xudong Zheng <7pkvm5aw@slicealias.com>

soburi commented:
@yonsch 
Could you take a look, please?
